File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/response/OrderResponse.java
package br.com.edu.order.infra.rest.v1.response;

import br.com.edu.order.domain.Order;
import br.com.edu.order.domain.Status;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderResponse {

    private UUID id;

    private Status status;

    private LocalDateTime created;

    @JsonProperty("created_by")
    private String createdBy;

    private LocalDateTime updated;

    @JsonProperty("updated_by")
    private String updatedBy;

    private BigDecimal total;

    private List<ItemResponse> items;

    public static OrderResponse fromModel(final Order order) {
        return OrderResponse.builder()
                .id(order.getId())
                .status(order.getStatus())
                .created(order.getCreated())
                .createdBy(order.getCreatedBy())
                .updated(order.getUpdated())
                .updatedBy(order.getUpdatedBy())
                .total(order.getTotal())
                .items(order.getItems().stream().map(ItemResponse::fromModel).toList())
                .build();
    }

}

File: order.service/order-service-app/src/main/kotlin/order/infra/rest/v1/response/OrderResponse.kt
package order.infra.rest.v1.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import order.domain.Order
import order.domain.Status
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
data class OrderResponse(
    val id: UUID? = null,

    val status: Status? = null,

    val created: LocalDateTime? = null,

    @JsonProperty("created_by")
    val createdBy: String? = null,

    val updated: LocalDateTime? = null,

    @JsonProperty("updated_by")
    val updatedBy: String? = null,

    val total: BigDecimal? = null,

    val items: List<ItemResponse>? = null
) {

    companion object {
        fun fromModel(order: Order): OrderResponse = OrderResponse(
            id = order.id,
            status = order.status,
            created = order.created,
            createdBy = order.createdBy,
            updated = order.updated,
            updatedBy = order.updatedBy,
            total = order.total,
            items = order.items.stream().map(ItemResponse::fromModel).toList()
        )
    }
}
